WooCommerce/Classes/ViewModels/MainTabViewModel.swift

```diff
@@ -38,6 +38,8 @@ final class MainTabViewModel {
 
     private var cancellables = Set<AnyCancellable>()
 
+    let tapToPayBadgePromotionChecker: TapToPayBadgePromotionChecker = TapToPayBadgePromotionChecker()
```

Comment:
Perhaps for another PR: Should we abstract this class into some `PromotionChecker` protocol to keep it detached from the specific TTP promotion checker? It could make sense also inject the protocol in the initializer for testability and to use this common protocol in the future for other feature announcements.